pagefaultgames/pokerogue
# [Bug][UI] Fixes to Run History Logging

## What are the changes the user will see?
Run History will reflect the state of held items at the start of the wave, not the end of the wave <-- a popular player request.
Run History will also display losses caused from trainer battles in Mystery Encounters correctly. 
Certain edge case losses will also have proper handling as well. 
Additionally, some Biomes in certain languages are too long for the first line. This causes problems in Session Preview. The line will be moved downwards as a result. 

## Why am I making these changes?
Addressing issues found here <-- https://github.com/pagefaultgames/pokerogue/issues/4706

## What are the changes from a developer perspective?
New function that gets the desired information to create a sessionSaveData object that fits Run History's needs.

Balance Approved --> Mysterious Chest requires at least two valid Pokemon in the party before spawning. 
Balance Approved (01/11/2024) --> Fiery Fallout + Lost at Sea MEs will not faint Pokemon in the party and will set Pokemon to an HP of 1 instead. 

### Screenshots/Videos
![image](https://github.com/user-attachments/assets/f4e80fe5-549a-4e55-823c-b4fbf27072ca)

## How to test the changes?
- Case 1 - Lose in a trainer battle within a Mystery Encounter
- Case 2 - Lose or use items during the battle and lose the battle
- Case 3 - Check the session preview of a run that is in a Biome with a very long name.

## Checklist
